open-craft/edx-analytics-pipeline
# Internal review: User Profiles Table in MySQL

Ticket: [OC-763](https://tasks.opencraft.com/browse/OC-763)

In order to support user-specific reports in the Insights dashboard for instances that want to do so, this PR takes the first step by making user data available within the MySQL DB. [A related PR](https://github.com/open-craft/edx-analytics-data-api/pull/1) to the data API will allow analytics software to access the data, and a final PR for the Dashboard will display it to the user.

**Changes**:
- Add user's "name" to the existing task that imports data from the LMS `auth_userprofile` table to Hive.
- Create a new `user_profile` table in the analytics MySQL DB. This new table contains basic user information (ID, name, email, username, gender, login date) extracted from the `auth_user` and `auth_userprofile` tables.

**Test Instructions**:
1. Setup an [analytics devstack](https://github.com/open-craft/edx-analytics-devstack). 
2. Modify your local LMS devstack to work with the new analytics devstack (See analytics devstack README).
3. (Optional:) run the various test suites and pipeline tests described in the analytics devstack README.
4. Login to the analytics devstack as the `analytics` user.
5. Make sure HDFS is running (e.g. try `hdfs dfs -ls /` or go to http://192.168.33.11:50070/ ). If not, see instructions to start it [here](https://github.com/open-craft/edx-analytics-devstack#testing-the-pipeline-answer-distribution).
6. Run the command:

```
   launch-task InsertToMysqlUserProfilesTask --local-scheduler
```
1. Wait for it to complete. Check for errors in the output. If there are some, check the analytics devstack README for possible remedies or try `vagrant provision` then try again.
2. To check if it worked, run `mysql -u root analytics --execute="SELECT * FROM user_profile;"`. You should get a result set showing all the users on your LMS devstack.

**TODO**:
- Discuss with upstream to find out if they're interested in accepting this code or not.
- Figure out how to write/run acceptance tests.
